### PR TITLE
VylBot 1.2

### PR DESCRIPTION
**Pull request by Vylpes**
Changes in this Pull Request
- The bot now DMs the user for reasons for their mute, ban, or kick
- The bot now has an announce command
- The bot now has a poll command

### CODE REVIEW

**Comment by Simyon264** [comment]
The Announce command will send an empty embed if no input is given.

File: commands/announce.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const discord = require('discord.js');
+const config = require('../config.json');
+const functions = require('../functions.js');
+
+var colourInfo = config.messageColours.info;
+var colourWarn = config.messageColours.warn;
+
+exports.run = function(message, prefix, args) {
+    if (message.member.roles.find(role => role.name == config.roles.moderator)) {
```
Comment (VylpesTester) [inline commands/announce.js:9]: This should give an error if no parameter is given

File: commands/ban.js [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
-	                functions.embed(message.channel, "", colourInfo, user.tag + " has been banned");
-	                functions.embed(message.guild.channels.find(channel => channel.name == config.channels.logging), "Member Banned", colourInfo, "Member: " + user.tag + "\n Reason: " + reason + "\n Moderator: " + message.author.tag);
-	            }).catch(err => {
-					functions.embed(message.channel, "", colourWarn, "There was an error banning this user, maybe I'm missing permissions?");
-					console.log(err);
-	            });
+				var reason = argsReason.join(" ");
+				if(message.guild.available){
+				var embed = new discord.RichEmbed()
+						.setTitle(`You were banned from: ${serverName}`)
```
Comment (VylpesTester) [inline commands/ban.js:22]: Adjust this

File: commands/help.js [hunk 3]
```diff
@@ -8,10 +8,12 @@ exports.run = function(message, prefix, args) {
 	var commands = "";
 	
 	commands += prefix + "about\n";
+	commands += prefix + "announce [title];[description];[url];[thumbnail]\n";
```
Comment (VylpesTester) [inline commands/help.js:11]: Once you fix the requirement of one parameter adjust this line here

File: commands/kick.js [hunk 4]
```diff
@@ -15,15 +17,26 @@ exports.run = function(message, prefix, args) {
 	            var argsReason = args;
 	            argsReason.splice(0, 2);
 	            
-	            var reason = argsReason.join(" ");
-	
-	            member.kick(reason).then(() => {
+				var reason = argsReason.join(" ");
+				if(message.guild.available){
+				var embed = new discord.RichEmbed()
```
Comment (VylpesTester) [inline commands/kick.js:22]: Adjust this

File: commands/mute.js [hunk 5]
```diff
@@ -16,16 +18,26 @@ exports.run = function(message, prefix, args) {
 	            argsReason.splice(0, 2);
 	
 	            var reason = argsReason.join(" ");
-	
+				if(message.guild.available){
 	            var mutedRole = message.guild.roles.find(role => role.name == config.roles.muted);
-				
+				var embed = new discord.RichEmbed()
```
Comment (VylpesTester) [inline commands/mute.js:23]: Adjust this

**Pull request merged**